src/cmd/run.js

```diff
@@ -63,23 +68,48 @@ export function defaultWatcherCreator(
     sourceDir,
     artifactsDir,
     onChange: () => {
-      log.debug(`Reloading add-on ID ${addonId}`);
-
-      return client.reloadAddon(addonId)
-        .catch((error) => {
-          log.error('\n');
-          log.error(error.stack);
-          desktopNotifications({
-            title: 'web-ext run: error occured',
-            message: error.message,
-          });
-          throw error;
-        });
+      return addonReload({addonId, client});
     },
     shouldWatchFile: (file) => fileFilter.wantFile(file),
   });
 }
 
+export type ReloadParams = {
+  addonId: string,
+  client: RemoteFirefox,
+  desktopNotifications?: typeof defaultDesktopNotifications,
+};
+
+export function defaultAddonReload(
+  {
+  addonId, client, desktopNotifications = defaultDesktopNotifications,
+ }: ReloadParams
+): Promise<void> {
+  log.debug(`Reloading add-on ID ${addonId}`);
+  return client.reloadAddon(addonId)
+    .catch((error) => {
+      log.error('\n');
+      log.error(error.stack);
+      desktopNotifications({
+        title: 'web-ext run: error occured',
+        message: error.message,
+      });
+      throw error;
+    });
+}
+
+export type ExitParams = {
+  logger?: typeof log,
+  killer?: typeof process.kill,
+};
+
+export async function defaultExitProgram(
+  {logger = log, killer = process.kill}: ExitParams = {},
+): Promise<void> {
+  logger.info('Exiting web-ext on user request');
+  killer(process.pid, 'SIGINT');
```

Comment:
@rpl I am in no way an expert on node, however, after you pointed out that `process.exit()` wasn't the best strategy, I looked for some answers and it seems, that sending `'SIGINT'` allows graceful shutdown and it also allows us to `process.on('SIGINT', function () {...}` in case we want to clean up something which is not taken care of. I am open to any suggestion and better sources of course!